### ISSUE TITLE
indent rule is changed in v3.2.0 breaking backwards compatibility

### ISSUE THREAD

**Issue by epoberezkin** [eslint/eslint#6803]
**What version of ESLint are you using?**
3.2.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

```json
{
    "rules": {
        "indent": [ 2, 2, {"SwitchCase": 1} ],
        "no-trailing-spaces": 2,
        "quotes": [ 2, "single", "avoid-escape" ],
        "linebreak-style": [ 2, "unix" ],
        "semi": [ 2, "always" ],
        "valid-jsdoc": [ 2, { "requireReturn": false } ],
        "no-invalid-this": 2,
        "no-console": [ 2, { "allow": ["warn", "error"] } ],
        "block-scoped-var": 2,
        "complexity": [2, 13],
        "curly": [2, "multi-or-nest", "consistent"],
        "dot-location": [2, "property"],
        "dot-notation": 2,
        "no-else-return": 2,
        "no-eq-null": 2,
        "no-fallthrough": 2,
        "no-return-assign": 2,
        "strict": [2, "global"],
        "no-shadow": 1,
        "no-use-before-define": [2, "nofunc"],
        "callback-return": 2
    },
    "env": {
        "node": true,
        "browser": true
    },
    "extends": "eslint:recommended"
}
```

**What did you do? Please include the actual source code causing the issue.**

fails indent rule:

```javascript
  return str.replace(SINGLE_QUOTE, '\\$&')
            .replace(/\n/g, '\\n') // eslint "indent" error in this and following lines
            .replace(/\r/g, '\\r')
            .replace(/\f/g, '\\f')
            .replace(/\t/g, '\\t');
```

```javascript
  return checkNotValid(p)
  .then(function (err) { // eslint "indent" error in this line
  });
```

```javascript
  return p.then(function (/* valid */) {
  })
  .catch(function (err) { // eslint "indent" error in this line
  });
```

```javascript
        fullValidate.errors
        .filter(function(err) { /**/ }) // eslint "indent" error in this and following lines
        .map(function(err) { /**/ })
        .forEach(function(p) { /**/ });
```

**What did you expect to happen?**

Pass indent rule, as it did in version 3.1.1

**What actually happened? Please include the actual, raw output from ESLint.**

```
115:13  error  Expected indentation of 4 space characters but found 12   indent
116:13  error  Expected indentation of 4 space characters but found 12   indent
117:13  error  Expected indentation of 4 space characters but found 12   indent
118:13  error  Expected indentation of 4 space characters but found 12   indent
```

```
425:3  error  Expected indentation of 4 space characters but found 2  indent
```

```
451:3  error  Expected indentation of 4 space characters but found 2  indent
```

```
86:9  error  Expected indentation of 10 space characters but found 8  indent
87:9  error  Expected indentation of 10 space characters but found 8  indent
88:9  error  Expected indentation of 10 space characters but found 8  indent
```

**Comment by mysticatea** [comment]
Thank you for this issue.

This is a duplicate of #6795.
Please track the issue.
